### ISSUE TITLE
Latest BL2 and TPS UHD patch update broke the autoexec

### ISSUE THREAD

**Issue by DesireForHappiness** [c0dycode/BL-AutoexecDLL#7]
Please introduce a new version of the autoexec. The latest patch seems to have broke it. I can still Launch the mods manually with console command but very much prefer your autoexec in case I forgot to open console..

The plugin folder and ddraw.dll also has to be removed from folder now or the game will have problem launching.

**Comment by c0dycode** [comment]
Yeah, I'm aware of the dll's needing to be updated. Unfortunately I'm still busy working out some last bugs with the Borderlands Hex Multitool. Once that's done, I can focus on the Plugins :)

**Comment by c0dycode** [comment]
I've updated the BL2 zip with a beta version for the UHD Update. TPS will follow in a bit.

Make sure to delete/rename the old one and use the autoexec-UHD.dll instead :)

As stated in the readme, consider this a beta with potential issues :)

**Comment by Sevin7** [comment]
I've tried the new DLL with no success. I got the BL2 update after I read some about how people were having issues with launching the game with mods, so when mine didn't launch I renamed my autoexec.txt to something else and it worked fine and I could execute my mods patch from the console in-game. Came here and found the new DLL and still have the same problem. I get to the launch menu but when I click "Play" it closes and nothing else happens. Though with the new DLL a command prompt opens briefly saying "Console opened!" I think. 

Also, some of the wording I see on your new readmes makes it sound like the Bl2 update is optional? It's not optional, is it? I had a 1.3GB update for TPS and ~400MB for BL2. What exactly is the update? It doesn't automatically enable the 4K textures and stuff, right? That's separate from the base games.

**Comment by c0dycode** [comment]
Make sure you have any outdated dll's removed from the Plugin folder.
For example "autoexec.dll" or "CommandInjector.dll". Pretty much any DLL that does not have the "UHD" in it's name. Also the "autoexec.dll" which is included.

If you get the ConsoleWindow, download it again. That was a leftover from debugging.

And in theory you could not receive the update if you choose to not keep the game updated in steam, I guess.

**Comment by Sevin7** [comment]
> Make sure you have any outdated dll's removed from the Plugin folder.
> For example "autoexec.dll" or "CommandInjector.dll". Pretty much any DLL that does not have the "UHD" in it's name. Also the "autoexec.dll" which is included.
> 
> If you get the ConsoleWindow, download it again. That was a leftover from debugging.
> 
> And in theory you could not receive the update if you choose to not keep the game updated in steam, I guess.

I renamed the old autoexec and commandinjector dll's to "-old", wouldn't that also work? I'll redownload.

**Comment by c0dycode** [comment]
As long as they are no longer detected as DLL-Files (check the Type coloumn of your explorer window) that's fine. Otherwise the name itself doesn't matter. The PluginLoader will load any ".dll"-file that's inside the Plugin-folder.

**Comment by Sevin7** [comment]
> As long as they are no longer detected as DLL-Files (check the Type coloumn of your explorer window) that's fine. Otherwise the name itself doesn't matter. The PluginLoader will load any ".dll"-file that's inside the Plugin-folder.

Oh, I see. Moved them out. It does launch now and does appear to work, though I don't get a console printout that it executed the patch and it doesn't skip the movies. My video settings are also completely different than when I last launched, is that another common problem with this update?

**Comment by Sevin7** [comment]
Strike that, it does echo that sparkservice and my mods were executed, I just forgot it's a chat printout and not console.

**Comment by c0dycode** [comment]
I think most of the .ini files in `Documents\My Games\Borderlands 2\WillowGame\Config` have been overwritten by the update.

So to skip those movies, you'd have to edit those again.

Glad to hear that it's working though :)

**Comment by Sevin7** [comment]
> I think most of the .ini files in `Documents\My Games\Borderlands 2\WillowGame\Config` have been overwritten by the update.
> 
> So to skip those movies, you'd have to edit those again.
> 
> Glad to hear that it's working though :)

Ah, yes, I suppose I did disable those in the .inis. Do you know what exactly the update did? It wasn't the actual UHD update, was it? Don't you have to download that separately?

**Comment by c0dycode** [comment]
Yeah, my best guess is, that it simply adjusts some things to allow for bigger texture files. But we don't really know exactly what has changed.

**Comment by Sevin7** [comment]
> Yeah, my best guess is, that it simply adjusts some things to allow for bigger texture files. But we don't really know exactly what has changed.

Wow, just opted into the 4K DLC and it's almost 8GB. I suppose that's to be expected if they ended up quadrupling a lot of the texture sizes.

**Comment by pyromaniac420710** [comment]
so i alt tabed out and back in and it crashed and i got an autoexec.dmp in the plugin folder
[autoexec.dmp.txt](https://github.com/c0dycode/BL-AutoexecDLL/files/3061671/autoexec.dmp.txt)
i just added the .txt so i could upload it here

**Comment by c0dycode** [comment]
Thanks for the dmp :) Unfortunately it did not contain any useful information :( (Nothing on your end or the code. Just a bad time for the crash to occur). But I've identified a few areas that I can improve the code a bit on. I'll upload a new version, once I've checked over the code and made those improvements. 👍

**Comment by c0dycode** [comment]
[BL2 DLL-Autoexec.zip](https://github.com/c0dycode/BL-AutoexecDLL/files/3071269/BL2.DLL-Autoexec.zip)
[BLTPS DLL-Autoexec.zip](https://github.com/c0dycode/BL-AutoexecDLL/files/3071270/BLTPS.DLL-Autoexec.zip)
@pyromaniac420710 Hopefully these are a little more stable :)

**Comment by DesireForHappiness** [comment]
> [BL2 DLL-Autoexec.zip](https://github.com/c0dycode/BL-AutoexecDLL/files/3071269/BL2.DLL-Autoexec.zip)
> [BLTPS DLL-Autoexec.zip](https://github.com/c0dycode/BL-AutoexecDLL/files/3071270/BLTPS.DLL-Autoexec.zip)
> @pyromaniac420710 Hopefully these are a little more stable :)

I am wondering if these suitable for everyone else or only pyromaniac420710? As the files on the main page has yet to be updated.

**Comment by c0dycode** [comment]
They can be used by everyone :)
I've not updated the mainfiles yet, since I've changed a bit of code on this version and so far I was the only tester :)

**Comment by DesireForHappiness** [comment]
I just tried the new files. I crashed once for TPS, but the second time I launch again, it works so I guess it is working fine as long as it doesn't crash often.

**Comment by pyromaniac420710** [comment]
so far working good :)

**Comment by twardle** [comment]
Just in case others have issues, if you are having a large number of crashes on startup, try a computer restart.  I was able to launch the game once in every 5-10 runs, but after a restart I'm consistently able to start the game up with autoexec.

**Comment by pyromaniac420710** [comment]
[autoexec.dmp.txt](https://github.com/c0dycode/BL-AutoexecDLL/files/3087257/autoexec.dmp.txt)
I tried to get it to crash had like 10 other things running and tabbed out when it was in the middle of playing the audio when you first load opportunity, don't think this is even autoexec fault, but i got a dmp so its here if you want it. other than that it works real smoothly :)

**Comment by c0dycode** [comment]
Thanks for the dmp and sorry for the late reply. Been sick lately so I've taken a break.
I'll take a look at it when I'm fully recovered again :)

**Comment by pyromaniac420710** [comment]
Its all good, no rush. I got another one I didn't even know it crashed, other than the random one just like the one before the update, but that never left a dmp for me before
[autoexec.dmp.txt](https://github.com/c0dycode/BL-AutoexecDLL/files/3125940/autoexec.dmp.txt)
